app/workers/argo_sync_worker.rb

```diff
@@ -0,0 +1,81 @@
+class ArgoSyncWorker
+  include Sidekiq::Worker
+  sidekiq_options retry: false
+
+  def perform(helm_infrastructure_id)
+    infrastructure = HelmInfrastructure.find(helm_infrastructure_id)
+
+    if infrastructure.blank? || !infrastructure.active?
+      render json: {
+        success: false,
+        errors: ["Infrastructure not found"],
+        code: 404
+      }, status: :not_found and return
+    end
+
+    release_name = infrastructure.cluster_name
+    repository = Figaro.env.HELM_REPOSITORY.to_s
+    chart_name = Figaro.env.HELM_CHART_NAME.to_s
+    chart_version = Figaro.env.HELM_CHART_VERSION.to_s
+
+    configs = "Release name: #{release_name} \nChart name: #{chart_name} \nChart version: #{chart_version} \n"
+    stdin_data = YAML.dump(infrastructure.values)
+
+    invocation_info = (
+      <<~EOS
+        #{configs}
+        Input:
+        #{stdin_data}
+      EOS
+    ).strip
+
+    sleep Figaro.env.argocd_worker_sync_interval.to_i
```

Comment:
let's move this after `DEPLOYMENT_STARTED` so we know that the delay is due to the deployment, not before

Reply:
updated